ospfd/ospf_vty.c

```diff
@@ -3666,6 +3666,154 @@ static int show_ip_ospf_interface_common(struct vty *vty, struct ospf *ospf,
 	return CMD_SUCCESS;
 }
 
+static void show_ip_ospf_interface_traffic_sub(struct vty *vty,
+					       struct ospf_interface *oi,
+					       json_object *json_interface_sub,
+					       u_char use_json)
+{
+	if (use_json) {
+		json_object_int_add(json_interface_sub,
+			    "ifIndex",
+			    oi->ifp->ifindex);
+		json_object_int_add(json_interface_sub,
+			    "helloIn",
+			    oi->hello_in);
+		json_object_int_add(json_interface_sub,
+			    "helloOut",
+			    oi->hello_out);
+		json_object_int_add(json_interface_sub,
+			    "dbDescIn",
+			    oi->db_desc_in);
+		json_object_int_add(json_interface_sub,
+			    "dbDescOut",
+			    oi->db_desc_out);
+		json_object_int_add(json_interface_sub,
+			    "lsReqIn",
+			    oi->ls_req_in);
+		json_object_int_add(json_interface_sub,
+			    "lsReqOut",
+			    oi->ls_req_out);
+		json_object_int_add(json_interface_sub,
+			    "lsUpdIn",
+			    oi->ls_upd_in);
+		json_object_int_add(json_interface_sub,
+			    "lsUpdOut",
+			    oi->ls_upd_out);
+		json_object_int_add(json_interface_sub,
+			    "lsAckIn",
+			    oi->ls_ack_in);
+		json_object_int_add(json_interface_sub,
+			    "lsAckOut",
+			    oi->ls_ack_out);
+	} else {
+		vty_out(vty,
+			"%-10s %8u/%-8u %7u/%-7u %7u/%-7u %7u/%-7u %7u/%-7u\n",
+			oi->ifp->name, oi->hello_in,
+			oi->hello_out,
+			oi->db_desc_in, oi->db_desc_out,
+			oi->ls_req_in, oi->ls_req_out,
+			oi->ls_upd_in, oi->ls_upd_out,
+			oi->ls_ack_in, oi->ls_ack_out);
+	}
+}
+
+/* OSPFv2 Packet Counters */
+static int show_ip_ospf_interface_traffic_common(struct vty *vty,
+						 struct ospf *ospf,
+						 int argc, int display_once,
+						 struct cmd_token **argv,
+						 int iface_argv,
+						 u_char use_json)
```

Comment:
I think it would be cleaner to remove the `argc`, `argv` and `iface_argv` arguments and instead have a `ifname` argument in their place. Then if `ifname` is NULL, show all interfaces. On the call sites, you could use `argv_find` to find the interface name (if it's given) instead of passing hard-coded indexes like `5` or `7`.

Reply:
I second this opinion!